### PR TITLE
Issue #SB-18924 fix: refactoring logs for better debugging

### PR DESCRIPTION
**Pull request by swetha-aggidevara**

### CODE REVIEW

File: src/app/helpers/utilityService.js [hunk 1]
```diff
@@ -116,6 +116,32 @@ const logError = (req, err, msg) => {
...
+
+const getLogObj = (req, err, msg) => {
+  const logObj =  {
+    did: _.get(req, 'headers.x-device-id'),
+    ts: _.get(req, 'headers.ts'),
+    url: _.get(req, 'originalUrl'),
+    uid: _.get(req, 'headers.x-msgid'),
+    sid: _.get(req, 'sessionID'),
+    payload: (JSON.stringify(_.get(req, 'body')) && JSON.stringify(_.get(req, 'body')).length <=500) ? JSON.stringify(_.get(req, 'body')) : '',
+    msg: msg,
```
Comment (rajeshkumaravel) [inline src/app/helpers/utilityService.js:139]: - Use ternary operator
Ex: 
```msg: msg ? msg : 'Unavaialable';```

Also for 
```error: JSON.stringify(err) ? JSON.stringify(err) : 'Unavailable'```

File: src/app/proxy/proxyUtils.js [hunk 2]
```diff
@@ -23,6 +24,7 @@ const apiInterceptor = new ApiInterceptor(keyCloakConfig, cacheConfig)
 
 const decorateRequestHeaders = function () {
   return function (proxyReqOpts, srcReq) {
+    logDebug(proxyReqOpts, {}, ' decorateRequestHeaders() called');
```
Comment (rajeshkumaravel) [inline src/app/proxy/proxyUtils.js:27]: - Here `srcReq` is actual request; please verify.
If yes; log with `srcReq`

File: src/app/routes/learnerRoutes.js [hunk 3]
```diff
@@ -28,12 +28,12 @@ module.exports = function (app) {
       },
       userResDecorator: (proxyRes, proxyResData, req, res) => {
         try {
-          logger.info({msg: '/learner/portal/user/v1/update called'});
+            logDebug(req, err, {msg: '/learner/portal/user/v1/update called'});
```
Comment (rajeshkumaravel) [inline src/app/routes/learnerRoutes.js:31]: - `err` param is undefined here

File: src/app/server.js [hunk 4]
```diff
@@ -52,7 +52,7 @@ app.use(session({
   saveUninitialized: false,
   store: memoryStore
 }))
-
+app.use(bodyParser.json())
```
Comment (rajeshkumaravel) [inline src/app/server.js:55]: - Why this is introduced at the app level?

File: src/app/helpers/httpRequestHandler.js [hunk 5]
```diff
@@ -1,13 +1,13 @@
 const requestPromise = require('request-promise');
 const logger = require('sb_logger_util_v2');
-
+const {logInfo, logDebug, logErr} = require('./utilityService');
 const sendRequest = async (options) => {
-  logger.info({msg: 'httpRequestHandler:sendRequest fetching data for ' + options.url});
+  logDebug({},{}, `httpRequestHandler:sendRequest is called with ${JSON.stringify(options)}`);
```
Comment (rajeshkumaravel) [inline src/app/helpers/httpRequestHandler.js:5]: - log only url; do not log entire options

File: src/app/server.js (hunk 4 above)
Comment (swetha-aggidevara) [reply src/app/server.js:55]: This is i need to add the payload but there is no req.body. So i added this

File: src/app/helpers/httpRequestHandler.js (hunk 5 above)
Comment (rajeshkumaravel) [reply src/app/helpers/httpRequestHandler.js]: Change
``` is called with ${JSON.stringify(options)}```
to 
is called with options.url

**Pull request merged**